tree/bst.go

```diff
@@ -160,3 +160,117 @@ func (n *node) getBSTNodeWithValue(val int) *node {
 
 	return n.right.getBSTNodeWithValue(val)
 }
+
+// changes By Prerna 08/01/2022
+
+// 1. Is validate BST
+func (t *BinarySearchTree) ValidBST() {
+	if t.isBST() {
+		fmt.Println("Valid BST")
+
+	} else {
+		fmt.Println("Invalid BST")
+	}
+}
+
+func (t *BinarySearchTree) isBST() bool {
+	if t == nil {
+		return true
+	}
+	return t.root.chkNode(100, 100)
```

Comment:
Use math.MinInt and math.MaxInt instead of 100 here. Min should start from the minimum value of int and max should start from max value of integer.